### PR TITLE
fix(security): annotating the map asked you for a password

### PR DESCRIPTION
**Pull request by davidp57**
Two defects a DCS session found in half an hour, both older than the work that went looking for them.

## Annotating the map asked you for a password

Closes `FIX-SECURITY-BEFORE-RECOGNITION`. Testing the `_transport` fix in game, the refusal printed **three times**. The refusal was correct; the count was a second bug.

`dispatchMarker` called `isAllowed` **before** the handler said whether it recognised the text, so every handler whose tier the pilot lacked printed a refusal for a command it would never have handled — `veafRadio` and `veafMove` (both `SENIOR_PILOT`) plus `veafTransportMission`'s own direct call.

**The everyday case was worse than `_transport`.** `veafMarkers` forwards every marker carrying text, and the loop only stops when a handler *consumes* the event. So a pilot writing **"RDV ici"** got two *"give the L1 password"* messages — for annotating the map. That predates 2026-08-13: the dispatcher comes from `SECREV-2`.

Fixed at the cause rather than by deduplicating the messages, which would have left the log full of security failures for commands nobody attempted. `registerCommandHandler` takes the keyphrase each module already declares, and the dispatcher skips a handler before testing anything. A handler declaring none keeps today's behaviour, so the change is additive.

Four tests count security **calls** rather than messages — the message is the symptom, the call is the defect.

`veafTransportMission`'s own call stays: it *is* a `_transport` marker and the pilot lacks the tier, so one message is the intended outcome. Moving that module into the dispatcher would change when its security runs and what consumes the event — its own ticket, not a passenger here.

## `validate` let through a mission DCS refuses

`FIX-SCRATCH-MISSION-PLAYABLE` ticket 02. Thirty seconds after loading a prepared mission, DCS opened **CHANGING COALITIONS** with every country unassigned: `coalitions` (which countries a side owns) was empty while `coalition` (the units) was populated. Two tables describe the same fact, and filling only the second gives units in a side that does not exist.

The mission had **passed `validate` and built cleanly**.

`validate` now reports:

- **error** — a side holding units with no country assigned, naming the side and what to fix;
- **warning** — a mission with no player slot at all, which is legitimate for a server-side scenario or a template library and must never refuse a build.

The guard was written **before** either fix and failed on the session's own mission, which is what proves it measures something. It reports nothing on the five real missions in `test/veaf-tools/`.

## What that lot still owes, named rather than implied

- **`blank_mission` ships `coalitions` empty** while its own comment claims `add_group` fills it in. No line of the MCP writes that table — the `VMR-088` shape, work deferred to a place that never did it.
- **Nothing can create a player slot.** Flagged the same afternoon as something a mission maker will need for certain. `set_unit_properties` refusing `Client` stays correct; what is missing is an action that *creates* the slot, and its **air-start half needs none of ticket 09's parking data**.

## One decision worth review

The three quirk readers (`indexed`, `numeric_first`, `CATEGORIES`) moved from `veaf_mission_mcp.mission_table` to `veaf_libs.mission_table`, re-exported so no import changed. The validator needs the same dict-or-list quirk the MCP actions do, the dependency runs MCP → `veaf_libs`, and a second copy would have received half the fixes.

## Also

`DCS-SESSION-TODO` records item 0 as **verified in game**: the 90 localised F10 labels read correctly, so the release is no longer gated on them.

## Verification

- 36 Lua suites pass; stylua clean
- Python 81.33% against the 81 gate; ruff, mypy, `docs-check` clean
- Version 6.13.101 across the three manifests

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Ensure marker command handlers recognise marker text before enforcing security and extend mission validation to reject missions that DCS cannot load while warning about missions without player slots.

Bug Fixes:
- Prevent map annotations and non-command marker text from triggering spurious security checks and password prompts.
- Add validation to detect missions with units in coalitions that have no assigned countries so they no longer pass silently.

Enhancements:
- Introduce an optional keyphrase on command handlers so the dispatcher can filter handlers by marker text prior to security checks while preserving current behaviour for handlers without keyphrases.
- Centralise mission-table helper readers (categories and dict/list indexing) in a shared library and re-export them from the MCP module.
- Extend mission validation to warn when a mission has no player slot, preserving legitimate server/template scenarios.

Documentation:
- Add backlog and ticket documentation for the security-order fix and scratch mission playability guard, and record in-session verification of the French F10 menu labels.

Tests:
- Add Lua tests asserting that marker text recognition happens before security checks and that keyphrase filtering behaves correctly, including case handling and handlers without keyphrases.
- Add Python tests covering new mission playability validation rules for coalitions and player slots, exercising both list and dict mission-table shapes.

Chores:
- Bump project version to 6.13.101 and update backlog index to reference new security and mission-playability fixes.

### CODE REVIEW

File: src/scripts/veaf/veafCommands.lua [hunk 1]
```diff
@@ -123,7 +123,12 @@ veafCommands.commandHandlers = {}
...
-function veafCommands.registerCommandHandler(fn, priority, security)
+-- @param keyphrase string — OPTIONAL but strongly advised: the marker text this handler answers to
+--                 (e.g. "_move"). Without it the dispatcher must check this handler's security for
+--                 **every** marker carrying text, so a pilot annotating the map with "RDV ici" was
+--                 told to give the L1 password once per tiered handler. Declaring it lets the
+--                 dispatcher skip the handler before checking anything.
+function veafCommands.registerCommandHandler(fn, priority, security, keyphrase)
   assert(type(fn) == "function", "veafCommands.registerCommandHandler: fn must be a function")
   assert(type(priority) == "number", "veafCommands.registerCommandHandler: priority must be a number")
   assert(
```
[the remark this answers is not in the document]
Comment (davidp57) [reply src/scripts/veaf/veafCommands.lua:134]: Applied — and the empty-string case is the one that mattered: `find("", 1, true)` returns 1, so a handler declaring `""` would have matched every text **while looking like it filtered**. Same outcome as declaring none, but reached by accident rather than stated. Normalised at registration, with the reason in a comment so the next reader does not have to rediscover it.

The missing test is added too: a side **absent** from `coalitions` is a different input from a side present but empty — `assigned.get(side)` is None rather than an empty container — and a hand-edited mission declaring only red and neutrals reaches exactly that shape.

The PRD fragment is reworded rather than taking the suggestion verbatim: the work was not "deferred to the shape VMR-088", it was deferred to a place that never did it, which is the *same shape as* VMR-088.
